src/utils/pgwire/src/pg_extended.rs

```diff
@@ -123,16 +123,33 @@ impl PgStatement {
         self.row_description.clone()
     }
 
-    pub fn instance(&self, portal_name: String, params: &[Bytes]) -> PgPortal {
+    pub async fn instance<SM: SessionManager>(
+        &self,
+        session: Arc<SM::Session>,
+        portal_name: String,
+        params: &[Bytes],
+    ) -> Result<PgPortal, ()> {
         let statement = cstr_to_str(&self.query_string).unwrap().to_owned();
 
         if params.is_empty() {
-            return PgPortal {
+            let row_description =
+                if statement.starts_with("select") || statement.starts_with("SELECT") {
```

Comment:
I guess you should take the first 6 characters and case-insensitively compare them with "select", which means you can convert all characters to lower-case. I don't know if "Select" would work?